code/datums/wounds/bones.dm

```diff
@@ -0,0 +1,209 @@
+	. = ..()
+	update_inefficiencies()
+
+// TODO: limps are unique so if we fix one of two broken legs i think they'll be limp free, look into this
+/datum/wound/brute/bone/remove_wound()
+	if(current_limp)
+		QDEL_NULL(current_limp)
+	interaction_efficiency_penalty = 1 // also the ordering and such of this
+	. = ..()
+
+/datum/wound/brute/bone/proc/check_splint_factor(obj/item/I)
+	for(var/item_path in splint_items)
+		if(istype(I, item_path))
+			return splint_items[item_path]
+
+/datum/wound/brute/bone/proc/update_inefficiencies()
+	if(limb.body_zone in list(BODY_ZONE_L_LEG, BODY_ZONE_R_LEG))
+		if(splint_factor)
+			limp_slowdown = initial(limp_slowdown) * splint_factor
+		else
+			limp_slowdown = initial(limp_slowdown)
+		current_limp = victim.AddComponent(/datum/component/limp)
+	else if(limb.body_zone in list(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM))
+		if(splint_factor)
+			interaction_efficiency_penalty = 1 + ((interaction_efficiency_penalty - 1) * splint_factor)
+		else
+			interaction_efficiency_penalty = interaction_efficiency_penalty
+
+	limb.update_wounds()
+
+
+/datum/wound/brute/bone/moderate
+	name = "joint dislocation"
+	desc = "Patient's bone has been unset from socket, causing pain and reduced motor function."
+	treat_text = "Recommended application of bonesetter to affected limb, though manual relocation may suffice."
+	examine_desc = "is awkwardly jammed out of place"
+	occur_text = "jerks violently and becomes unseated"
+	severity = WOUND_SEVERITY_MODERATE
+	interaction_efficiency_penalty = 1.5
+	limp_slowdown = 3
+	threshold_minimum = 35
+	threshold_penalty = 15
+	treatable_tool = TOOL_BONESET
+
+/datum/wound/brute/bone/moderate/try_handling(mob/living/carbon/human/user)
+	if(user.pulling != victim || user.zone_selected != limb.body_zone || user.a_intent == INTENT_GRAB)
+		return FALSE
+
+	if(user.grab_state == GRAB_PASSIVE)
+		to_chat(user, "<span class='warning'>You must have [victim] in an aggressive grab to manipulate [victim.p_their()] [src]!</span>")
+		return TRUE
+
+	if(user.grab_state >= GRAB_AGGRESSIVE)
+		user.visible_message("<span class='danger'>[user] begins twisting and straining [victim]'s dislocated [limb.name]!</span>", "<span class='notice'>You begin twisting and straining [victim]'s dislocated [limb.name]...</span>", ignored_mobs=victim)
+		to_chat(victim, "<span class='userdanger'>[user] begins twisting and straining your dislocated [limb.name]!</span>")
+		if(user.a_intent == INTENT_HELP)
+			chiropractice(user)
+		else
+			malpractice(user)
+		return TRUE
+
+/datum/wound/brute/bone/moderate/proc/chiropractice(mob/living/carbon/human/user)
+	var/time = base_treat_time
+	var/time_mod = user.mind?.get_skill_modifier(/datum/skill/medical, SKILL_SPEED_MODIFIER)
+	var/prob_mod = user.mind?.get_skill_modifier(/datum/skill/medical, SKILL_PROBS_MODIFIER)
+	if(time_mod)
+		time *= time_mod
+
+	if(do_after(user, time, TRUE, victim))
```

Comment:
Early return.

Reply:
Could you clarify on this? The line below
`if(QDELETED(src) || !limb)
    return`
Checks to see if our wound or limb have been deleted in the time during the do_after, which I thought was standard procedure. Otherwise I'm not sure what you mean by early return.

Although perhaps it'd be cleaner if I just put those into a proc to use as a callback for the do_after itself, so I don't have to have those two ugly lines on every treatment do_after

Reply:
Early return means doing something like this:
```suggestion
	if(!do_after(user, time, TRUE, victim))
		return
```
Then indent all the rest with one tab less.
Makes sense?

Reply:
Oh, I wasn't aware you could do that with do_after, but it makes sense now that I think about it. Cheers